spec/models/redirection_spec.rb

```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+describe Redirection do
+  describe "associations" do
+    it { should belong_to(:next).
+                class_name("Redirection") }
+    it { should have_one(:previous).
+                class_name("Redirection").
+                with_foreign_key("next_id") }
+  end
+
+  describe "validations" do
+    it { should validate_presence_of :slug }
+    it { should validate_presence_of :url }
+    it { should validate_uniqueness_of(:next_id) }
```

Comment:
Oh whoops, I as doing `allow_blank` but it's [broken](https://github.com/thoughtbot/shoulda-matchers/issues/776)